### ISSUE TITLE
multi scale training perform badly when training from scratch

### ISSUE THREAD

**Issue by gaopeng-eugene** [DrSleep/tensorflow-deeplab-resnet#41]
I tried the multi scale training methods and found that the performance is much worse than the train.py when training from scratch.
How about your experiment?

**Comment by DrSleep** [comment]
I haven't run the experiments until the full convergence; @arslan-chaudhry might provide more information on this one.

**Comment by arslan-chaudhry** [comment]
I have run the **train_msc.py** script for 20k iterations and reported the results here: https://github.com/DrSleep/tensorflow-deeplab-resnet/issues/18. Can you please share the details of your experiment - initialization model, hyper parameters etc? Also, did you use random-scale and random-mirror options? If it helps, I can share my trained model somewhere?

**Comment by gaopeng-eugene** [comment]
this is my training code
CUDA_VISIBLE_DEVICES=5 python train_msc.py --random-scale --random-mirror --is-training --restore-from ../../tensorflow-deeplab-resnet3/deeplab_resnet_init.ckpt

I use the default parameter in your code.

**Comment by arslan-chaudhry** [comment]
Try running the training without *--is-training* flag. Original paper is not changing the batchnorm statistics.

**Comment by holyseven** [comment]
@arslan-chaudhry Hi, I read from [here](https://github.com/DrSleep/tensorflow-deeplab-resnet/issues/18#issuecomment-279721520) that you got 76.5% on val and 76.78% on test. Could you share the detail of your experiments? what are the hyperparameters you used? 

Because I tried training without msc, batch size 10, lr 2.5e-4, num_steps 20K, random_scale, random_crop, without is_training flag, the best iou I got is 75.3%. 
Training with msc, batch size 1, grad_update_every 10, lr 2.5e-4, num_steps 20K, random_scale, random_crop, without is_training flag, the best iou I got is 75.1%. 

(training set is about 10K images, validation set is about 1500 images, 'void' is also correctly ignored. restore-from deeplab_resnet_init.ckpt)

Did I miss something?

**Comment by arslan-chaudhry** [comment]
Did you try 'random-mirror' as well?
Also which evaluation script are you using? I calculated IOU with evaluate_msc.py checked in the master branch.

**Comment by holyseven** [comment]
@arslan-chaudhry Sorry... I made a mistake. I used random-scale and random-mirror (random-crop is set by default). And I used evaluate.py.
After changing to evaluate_msc.py, I get 76.5% too and I notice in the article, they got 76.35%. Excellent! Thanks for your help.

**Comment by John1231983** [comment]
Hi @holyseven  and @arslan-chaudhry: Thanks for your result. I also got 76.3% in validation set (SegmentationClassAug) using `python train_msc.py --random-mirror --random-scale`. I have three questions:

1. To achieve 76.78% on testset, do you merge train and val sets as training set and re-train again with `deeplab_resnet_init.ckpt`
2. Why is the validation result of multiple-scale (76%) lower than single-scale (86%)?
3. When you submit the result to PASCAL evaluation server, do you results that provided by `inference.py` or another script in testing set?

Thanks so much

**Comment by mokii** [comment]
I got 84% val accuracy by evaluate_msc.py training with train_msc.py. Is it rational?
